src/libaktualizr/aktualizr_api.h

```diff
@@ -151,6 +159,36 @@ typedef enum {
   AKTUALIZR_UPDATE_AWAITING_REBOOT
 } aktualizr_update_ui_category_t;
 
+#define AKTUALIZR_UI_STATUS_MAX_LEN 300
+
+/**
+ * A snapshot of the current update state
+ */
+struct aktualizr_update_state {
+  /**
+   * The high-level category of this update state snapshot.
+   */
+  aktualizr_update_ui_category_t ui_category;
+
+  /**
```

Comment:
Alternative: just put a big `char[]` (thoughtfully sized) and insert new fields there in new api versions while reducing the array's size. The only constraint to satisfy would be that `sizeof(struct aktualizr_update_state)` will stay constant and that old fields are not moved. It can be checked with a C11 `static_assert(sizeof(xxx) == 64)`

Also, maybe we could expose an api version just in case.